posthog/api/test/test_survey.py

```diff
@@ -403,6 +403,34 @@ def test_used_in_survey_is_populated_correctly_for_feature_flag_list(self) -> No
                 [("flag_0", []), (ff_key, [created_survey1, created_survey2])],
             )
 
+    def test_updating_survey_with_invalid_iteration_count_is_rejected(self):
+        survey_with_targeting = self.client.post(
+            f"/api/projects/{self.team.id}/surveys/",
+            data={
+                "name": "survey with targeting",
+                "type": "popover",
+                "targeting_flag_filters": {
+                    "groups": [
+                        {
+                            "variant": None,
+                            "rollout_percentage": None,
+                        }
+                    ]
+                },
+                "iteration_count": MAX_ITERATION_COUNT + 1,
+                "conditions": {"url": "https://app.posthog.com/notebooks"},
+            },
+            format="json",
+        )
+
+        assert survey_with_targeting.status_code == status.HTTP_400_BAD_REQUEST
+        assert survey_with_targeting.json() == {
+            "type": "validation_error",
+            "code": "max_value",
+            "detail": "Ensure this value is less than or equal to 500.",
```

Comment:
Nitpick, in case we ever change this value in the future, this will avoid us from having to change it here as well

```suggestion
            "detail": f"Ensure this value is less than or equal to {MAX_ITERATION_COUNT}.",
```